src/jit/morph.cpp

```diff
@@ -7043,31 +7249,145 @@ bool Compiler::fgCanFastTailCall(GenTreeCall* callee)
             }
             else
             {
-                hasMultiByteArgs = true;
+                hasMultiByteStackArgs = true;
             }
         }
+        else
+        {
+            varTypeIsFloating(argx) ? ++calleeFloatArgRegCount : ++calleeArgRegCount;
+        }
+
+#if !DEBUG
+        // We can break early on multiByte cases.
```

Comment:
Should be `#ifndef DEBUG`

However, it worries me that DEBUG and !DEBUG might be doing different work here. Why is that required?